internal/ledgerutil/identifytxs.go

```diff
@@ -0,0 +1,479 @@
+	snapshotInfo := blockchainInfo.GetBootstrappingSnapshotInfo()
+	if snapshotInfo != nil {
+		minBlockHeight = snapshotInfo.GetLastBlockInSnapshot() + uint64(1)
+	}
+	maxBlockHeight := blockchainInfo.GetHeight() - 1
+	maxTxHeight := uint64(math.MaxUint64)
+	// Check if compKeyMap's highest record is earlier than block store's max block height
+	if inputKeyMapWrapper.maxBlockNum <= maxBlockHeight {
+		maxBlockHeight = inputKeyMapWrapper.maxBlockNum
+		maxTxHeight = inputKeyMapWrapper.maxTxNum
+	}
+	blocksItr, err := blockStore.RetrieveBlocks(minBlockHeight)
+	if err != nil {
+		return 0, 0, err
+	}
+	defer blocksItr.Close()
+
+	inputKeyMap := inputKeyMapWrapper.compKeyMap
+	// Set block index to minimum allowed block height
+	blockIndex := minBlockHeight
+	txIndex := uint64(0)
+	// Iterate through all completed blocks of block store
+	for blockIndex <= maxBlockHeight {
+		// Unpack next block
+		nextBlock, err := blocksItr.Next()
+		if err != nil {
+			return 0, 0, err
+		}
+		block := nextBlock.(*common.Block)
+		blockData := block.GetData().GetData()
+
+		// Iterate through block transactions
+		for txIndexInt, envBytes := range blockData {
+			txIndex = uint64(txIndexInt)
+			// Extract next transaction
+			env, err := protoutil.GetEnvelopeFromBlock(envBytes)
+			if err != nil {
+				return 0, 0, err
+			}
+			pl, err := protoutil.UnmarshalPayload(env.GetPayload())
+			if err != nil {
+				return 0, 0, err
+			}
+			ch, err := protoutil.UnmarshalChannelHeader(pl.Header.ChannelHeader)
+			if err != nil {
+				return 0, 0, err
+			}
+			txID := ch.GetTxId()
+
+			// Extract write set from transaction then iterate through transaction write set
+			res, err := protoutil.GetActionFromEnvelope(envBytes)
+			if err != nil {
+				return 0, 0, err
+			}
+			// RW set
+			txRWSet := &rwsetutil.TxRwSet{}
+			err = txRWSet.FromProtoBytes(res.GetResults())
+			if err != nil {
+				return 0, 0, err
+			}
+			// Iterate namespaces in read write set
+			nsRWSets := txRWSet.NsRwSets
+			for _, nsRWSet := range nsRWSets {
+				namespace := nsRWSet.NameSpace
+				// Iterate public keys
+				writes := nsRWSet.KvRwSet.GetWrites()
+				for _, write := range writes {
+					key := write.GetKey()
+					ck := compKey{namespace: namespace, key: key}
+					_, exists := inputKeyMap[ck]
+					if exists {
+						// Get txValidationCode
+						txvCode, _, err := blockStore.RetrieveTxValidationCodeByTxID(txID)
+						if err != nil {
+							return 0, 0, err
+						}
+						// Create new transaction entry
+						entry := txEntry{
+							TxID:      txID,
+							BlockNum:  blockIndex,
+							TxNum:     txIndex,
+							TxVStatus: txvCode.String(),
+							KeyWrite:  string(write.GetValue()),
+						}
+						// Capture new transaction entry
+						ckMapEmpty, err := captureTx(inputKeyMap, ck, entry)
+						if err != nil {
+							return 0, 0, err
+						}
+						// No more keys left
+						if ckMapEmpty {
+							return minBlockHeight, blockIndex, nil
```

Comment:
In my trial I had several transactions for a key. But only one transaction gets written to txlist1.json. It appears that ckMapEmpty gets set to true after the first 'hit' and it returns here. Is this the intended design? Why return after the first hit? Don't we want to see all the transactions for the key?